src/utils/aspect-ratio-support.tsx

```diff
@@ -26,3 +27,92 @@ export const AspectRatioSupportProvider: React.FC<Props> = ({children}) => {
 };
 
 export const useSupportsAspectRatio = (): boolean => React.useContext(AspectRatioSupport);
+
+const useAspectRatioStyles = createUseStyles(() => ({
+    container: {
+        '@supports (aspect-ratio: 1 / 1)': {
+            aspectRatio: ({aspectRatio}) => aspectRatio ?? 'unset',
+        },
+        '$wrapper &': {
+            position: ({aspectRatio}) => (aspectRatio ? 'absolute' : 'static'),
+            width: '100%',
+            height: '100%',
+            top: 0,
+            left: 0,
+        },
+    },
+
+    wrapper: {
+        overflow: 'hidden',
+        maxWidth: '100%',
+        maxHeight: '100%',
+        position: 'relative',
+        paddingTop: ({aspectRatio, width}) => {
+            if (!aspectRatio) {
+                return 0;
+            }
+            if (width && typeof width === 'string' && width.endsWith('%')) {
+                return `${Number(width.replace('%', '')) / aspectRatio}%`;
+            }
+            return `${100 / aspectRatio}%`;
+        },
+    },
+}));
+
+type AspectRatioElementProps = {
+    width?: number | string;
+    height?: number | string;
+    aspectRatio: number;
+    children: React.ReactNode;
+    as?: React.ComponentType<any> | string;
+    style?: React.CSSProperties;
+    className?: string;
+};
+
+export const AspectRatioElement = (props: AspectRatioElementProps): JSX.Element => {
+    const supportsAspectRatio = useSupportsAspectRatio();
+    const withCssAspectRatio =
+        typeof props.width !== 'number' && typeof props.height !== 'number' && props.aspectRatio !== 0;
```

Comment:
not an expert at all about the aspect ratio, so asking to make sure I got this: in case relative measures are set for both height and width, and aspectRatio is set: the height will be ignored in order to follow the ratio set? Did I get right?

Reply:
I lost this explanation

```  
// if width or height are numeric, we can calculate the other with the ratio without css.
        // if aspect ratio is 0, we use the original image proportions

Reply:
wat